ttnn/cpp/ttnn/operations/embedding/embedding.hpp

```diff
@@ -30,31 +30,38 @@ struct EmbeddingOperation {
         if (pad_token.has_value()) {
             embeddings_type = EmbeddingsType::PADDED;
         }
+        Tensor mutable_input_tensor = input_tensor_arg;
+        Tensor mutable_weight = weight_arg;
+        if (mutable_input_tensor.get_layout() == ttnn::TILE_LAYOUT) {
+            mutable_input_tensor = ttnn::to_layout(mutable_input_tensor, ttnn::ROW_MAJOR_LAYOUT, std::nullopt, std::nullopt, (Device*)nullptr);
+        }
+        if (mutable_weight.get_layout() == ttnn::TILE_LAYOUT) {
+            mutable_weight = ttnn::to_layout(mutable_weight, ttnn::ROW_MAJOR_LAYOUT, std::nullopt, std::nullopt, (Device*)nullptr);
+        }
+        auto hidden_embedding_dim = mutable_weight.get_shape()[-1];
+        auto padded_hidden_embedding_dim = mutable_weight.get_shape().with_tile_padding()[-1];
+        auto weight = ttnn::unsqueeze_to_4D(mutable_weight);
 
-        auto hidden_embedding_dim = weight_arg.get_shape()[-1];
-        auto padded_hidden_embedding_dim = weight_arg.get_shape().with_tile_padding()[-1];
-        auto weight = ttnn::unsqueeze_to_4D(weight_arg);
-
-        auto batch_size = input_tensor_arg.get_shape()[0];
-        auto sentence_size = input_tensor_arg.get_shape()[-1];
+        auto batch_size = mutable_input_tensor.get_shape()[0];
+        auto sentence_size = mutable_input_tensor.get_shape()[-1];
         auto input_tensor =
-            ttnn::reshape(input_tensor_arg, ttnn::Shape{std::array<uint32_t, 4>{batch_size, 1, 1, sentence_size}});
+            ttnn::reshape(mutable_input_tensor, ttnn::Shape{std::array<uint32_t, 4>{batch_size, 1, 1, sentence_size}});
 
-        bool tilized = layout == ttnn::TILE_LAYOUT;
```

Comment:
Curious about the perf implications of removing the fused tilize atm. We can see what that's like by running perf post commit

Reply:
To my understanding, embedding currently really only supports fused tilized if the input layout is tiled. However, currently the operation also only works when the input is row_major. Considering we are only converting to RM, the input will never be tiled and so we set 'tilized' to always be false. But, granted, I don't know how fuzed tilize works yet.

Reply:
```
if (this->tilized) {
        TT_FATAL(a.get_legacy_shape()[-1] % TILE_HEIGHT == 0, "Error");
        TT_FATAL(weights.get_legacy_shape()[-1] % TILE_WIDTH == 0, "Number of columns in table must be factor of tile width");
    }
```
At least according to the assertion, it doesn't require input or weights to be tilized, it requires input width and weight width to be a multiple of tile dims but still row major. Granted I don't know for sure embeddings is currently working properly or not, but it seems like you can change the condition to match the assert and run fused tilize for these cases instead of forcing it to always be false.

Reply:
Hmm the intention way back when i added it was to support row major input but Tilize on the output.

Reply:
That's still the use case. Right now input only supports row major. These conditions are that if we want to tilize the output the resulting output shape must be tile sized.